### PR TITLE
add a template for pull request

### PR DESCRIPTION
**Pull request by SarahAlidoost**
It adds a PR template containing a list of tasks.  It closes https://github.com/ESMValGroup/tutorial/issues/18

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,9 +1,26 @@
...
+**Tasks**
 
-Please keep in mind that lesson maintainers are volunteers and it may be some time before they can respond to your contribution. Although not all contributions can be incorporated into the lesson materials, we appreciate your time and effort to improve the curriculum.  If you have any questions about the lesson maintenance process or would like to volunteer your time as a contribution reviewer, please contact The Carpentries Team at team@carpentries.org.  
+-   [ ] Read [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md).
+-   [ ] Create an [issue](https://github.com/ESMValGroup/tutorial/issues) to discuss what you are going to do.
+-   [ ] Give this pull request a descriptive title.
+-   [ ] If you are contributing to the lesson materials, make sure the content does not have spelling or grammar errors.
+-   [ ] Please use `markdownlinter` to check that your files do not contain errors.
```
Comment (JaroCamphuijsen) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
-   [ ] Please use `markdownlint` to check that your files do not contain errors.
```

**Comment by JaroCamphuijsen** [comment]
When I ran the PR template through a markdownlint tool I got: <details><summary> these comments</summary>

21 - MD012 Multiple consecutive blank lines [Expected: 1; Actual: 2]
9 - MD030 Spaces after list markers [Expected: 1; Actual: 3]
10 - MD030 Spaces after list markers [Expected: 1; Actual: 3]
11 - MD030 Spaces after list markers [Expected: 1; Actual: 3]
12 - MD030 Spaces after list markers [Expected: 1; Actual: 3]
13 - MD030 Spaces after list markers [Expected: 1; Actual: 3]
14 - MD030 Spaces after list markers [Expected: 1; Actual: 3]
18 - MD030 Spaces after list markers [Expected: 1; Actual: 3]
19 - MD030 Spaces after list markers [Expected: 1; Actual: 3]
7 - MD036 Emphasis used instead of a heading [Context: "Tasks"]
</details>

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,9 +1,26 @@
...
+-   [ ] Preview changes on your own machine before pushing them to GitHub by running `make docker-serve`.
 
----
+New episode
```
Comment (JaroCamphuijsen) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
## New episode
```

**Comment by JaroCamphuijsen** [comment]
Looks good to me. What do you think @ledm , let's merge?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,9 +1,25 @@
...
+## Tasks
 
-Please keep in mind that lesson maintainers are volunteers and it may be some time before they can respond to your contribution. Although not all contributions can be incorporated into the lesson materials, we appreciate your time and effort to improve the curriculum.  If you have any questions about the lesson maintenance process or would like to volunteer your time as a contribution reviewer, please contact The Carpentries Team at team@carpentries.org.  
+- [ ] Read [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md).
+- [ ] Create an [issue](https://github.com/ESMValGroup/tutorial/issues) to discuss what you are going to do.
+- [ ] Give this pull request a descriptive title.
+- [ ] If you are contributing to the lesson materials, make sure the content does not have spelling or grammar errors.
+- [ ] Please use `markdownlint` to check that your files do not contain errors.
```
Comment (ledm) [inline .github/PULL_REQUEST_TEMPLATE.md]: Can you give the explicit command here? 
Also, do we need to add `markdownlint` to the conda environment.yml?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,9 +1,25 @@
...
+* * *
 
-Thanks for contributing! If this contribution is for instructor training, please send an email to checkout@carpentries.org with a link to this contribution so we can record your progress. You’ve completed your contribution step for instructor checkout just by submitting this contribution.  
+## Tasks
 
-Please keep in mind that lesson maintainers are volunteers and it may be some time before they can respond to your contribution. Although not all contributions can be incorporated into the lesson materials, we appreciate your time and effort to improve the curriculum.  If you have any questions about the lesson maintenance process or would like to volunteer your time as a contribution reviewer, please contact The Carpentries Team at team@carpentries.org.  
+- [ ] Read [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md).
+- [ ] Create an [issue](https://github.com/ESMValGroup/tutorial/issues) to discuss what you are going to do.
+- [ ] Give this pull request a descriptive title.
+- [ ] If you are contributing to the lesson materials, make sure the content does not have spelling or grammar errors.
```
Comment (ledm) [inline .github/PULL_REQUEST_TEMPLATE.md]: Can we recommend a spellcheck app for markdown?

**Comment by ledm** [comment]
Looks good to me, only very minor comments. Nice work.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (JaroCamphuijsen) [reply .github/PULL_REQUEST_TEMPLATE.md]: Yeah that sounds like a good plan for tutorials in any case!

Comment (SarahAlidoost) [reply .github/PULL_REQUEST_TEMPLATE.md]: @ledm Thanks. I could install Grammarly in VS Code. It is not very good as it should be. Shall suggest this?

**Comment by ledm** [comment]
We probably want to add a list of bullet points for PRs containing new lessons too:

- [ ] Lesson is clearly explained, and includes questions and objectives. 
- [ ] All code instructions have been tested.
- [ ] Lesson is a reasonable length. If its very long, please consider splitting it into two parts.
- [ ] Lesson ends with a summary of the key points. 

Just some ideas, but think it would be useful.

**Comment by SarahAlidoost** [comment]
@JaroCamphuijsen  @ledm Thanks for your comments. 
There are not any recommendations regarding a linter and spell/grammar checker by Carpentry.

Regarding the linter tool for example, in vscode, a markdownlint can be installed as an extension. Users don’t need to run a command line for that. However, none-vscode users can install another tool via `gem install mdl,` and run `mdl markdown_filename`. 
Regarding the spell/grammar checker, an option is Grammarly vscode extension. It depends on the editors that is used to develop lesson material.  I will ask other colleagues for more suggestions.
The contributing.md is the best place to explain about both linter and spell/grammar checker. I update this PR template accordingly.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (SarahAlidoost) [reply .github/PULL_REQUEST_TEMPLATE.md]: @ledm please see my [comment](https://github.com/ESMValGroup/tutorial/pull/19#issuecomment-601694273) . Fo second question, yes, please.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (SarahAlidoost) [reply .github/PULL_REQUEST_TEMPLATE.md]: please see my [comment](https://github.com/ESMValGroup/tutorial/pull/19#issuecomment-601694273) .

**Comment by SarahAlidoost** [comment]
Thanks a lot. Indeed very useful items. Please see below: 
> We probably want to add a list of bullet points for PRs containing new lessons too:
> 
> * [ ]  Lesson is clearly explained, and includes questions and objectives.
**it will be covered in the `lesson formatting` section in `contributing.md`**
> * [ ]  All code instructions have been tested.
**I added this item to the PR template.**
> * [ ]  Lesson is a reasonable length. If its very long, please consider splitting it into two parts.
**it will be covered in the `lesson formatting` section in `contributing.md`**
> * [ ]  Lesson ends with a summary of the key points.
**it will be covered in the `lesson formatting` section in `contributing.md`**
> 
> Just some ideas, but think it would be useful.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -1,9 +1,27 @@
...
+* * *
 
-Please keep in mind that lesson maintainers are volunteers and it may be some time before they can respond to your contribution. Although not all contributions can be incorporated into the lesson materials, we appreciate your time and effort to improve the curriculum.  If you have any questions about the lesson maintenance process or would like to volunteer your time as a contribution reviewer, please contact The Carpentries Team at team@carpentries.org.  
+## Tasks
 
----
+- [ ] Read [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md).
+- [ ] Create an [issue](https://github.com/ESMValGroup/tutorial/issues) to discuss what you are going to do.
+- [ ] Give this pull request a descriptive title.
+- [ ] If you are contributing to the lesson materials, make sure the content does not have spelling or grammar errors. See `Lesson development` in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md).
```
Comment (ledm) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] If you are contributing to the lesson materials, please make sure the content conforms to the `Lesson development` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors. 
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -1,9 +1,27 @@
...
+- [ ] Please use a `markdown lint tool` to check that your files do not contain errors. See `Lesson development` in[CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md).
```
Comment (ledm) [inline .github/PULL_REQUEST_TEMPLATE.md]: We already suggest checking the contributing.md elsewhere, don't need it twice. 

```suggestion
- [ ] Please use a `markdown lint tool` to check that your markdown files do not contain errors, bugs, or stylistic errors.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -1,9 +1,27 @@
...
+- [ ] Preview changes on your own machine before pushing them to GitHub by running `make serve`, alternatively `make docker-serve`. See `Lesson development` in[CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md).
```
Comment (ledm) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] Preview changes on your own machine before pushing them to GitHub by running `make serve`, alternatively `make docker-serve`. Please see the `Previewing your changes locally` section in[CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) for installation instructions.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 8]
```diff
@@ -1,9 +1,27 @@
...
+- [ ] All code instructions have been tested.
```
Comment (ledm) [inline .github/PULL_REQUEST_TEMPLATE.md]: Do we want to make sure that these have been tested by the author and the reviewer?
```suggestion
- [ ] All code instructions have been tested by the PR author.
- [ ] All code instructions have been tested by the reviewer.
```

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (bouweandela) [reply .github/PULL_REQUEST_TEMPLATE.md]: I would recommend using a markdown linter that is supported by an online service that can be integrated with the repository. Since we use Codacy for ESMValCore and ESMValTool as well, that would be a familiar option, it supports [remark-lint](https://www.google.com/search?q=remarklint).

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 8 above)
Comment (SarahAlidoost) [reply .github/PULL_REQUEST_TEMPLATE.md]: @ledm Thanks. I think that by default both reviewers and authors should check all the items provided in this template. So, we don't need to mention `reviewer` or `author` for this specific item.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 9]
```diff
@@ -1,9 +1,27 @@
...
 
-Please keep in mind that lesson maintainers are volunteers and it may be some time before they can respond to your contribution. Although not all contributions can be incorporated into the lesson materials, we appreciate your time and effort to improve the curriculum.  If you have any questions about the lesson maintenance process or would like to volunteer your time as a contribution reviewer, please contact The Carpentries Team at team@carpentries.org.  
+## Tasks
 
----
+- [ ] Read [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md).
+- [ ] Create an [issue](https://github.com/ESMValGroup/tutorial/issues) to discuss what you are going to do.
+- [ ] Give this pull request a descriptive title.
+- [ ] If you are contributing to the lesson materials, please make sure the content conforms to the `Lesson development` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors. 
+- [ ] Please use a `markdown lint tool` to check that your markdown files do not contain errors, bugs, or stylistic errors.
```
Comment (JaroCamphuijsen) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] Please use a `markdownlint` tool to check that your markdown files do not contain errors, bugs, or stylistic errors.
```

Comment (JaroCamphuijsen) [reply .github/PULL_REQUEST_TEMPLATE.md]: I would use either "`markdownlint` tool" or just "markdown lint tool" without monospace font

**Comment by JaroCamphuijsen** [comment]
Apart from the tiny comment I left, I think this is ready to merge!

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (SarahAlidoost) [reply .github/PULL_REQUEST_TEMPLATE.md]: @bouweandela thanks. I applied your suggestion here based on the same task in ESMValTool PR template.

**Comment by SarahAlidoost** [comment]
@ledm @JaroCamphuijsen I think the template is ready, Shall we merge this PR?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 10]
```diff
@@ -1,9 +1,27 @@
...
+## Tasks
 
----
+- [ ] Read [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md).
+- [ ] Create an [issue](https://github.com/ESMValGroup/tutorial/issues) to discuss what you are going to do.
+- [ ] Give this pull request a descriptive title.
+- [ ] If you are contributing to the lesson materials, please make sure the content conforms to the `Lesson development` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors.
+- [ ] Preferably Codacy code quality checks pass. However, a few remaining hard to solve Codacy issues are still acceptable. Status can be seen below your pull request. If there is an error, click the link to find out why. If you suspect Codacy may be wrong, please ask by commenting.
```
Comment (bouweandela) [inline .github/PULL_REQUEST_TEMPLATE.md:14]: This seems a bit confusing, as this repository does not contain code? Or does it?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 11]
```diff
@@ -1,9 +1,27 @@
-Please delete this line and the text below before submitting your contribution.
+# Pull Request checklist
 
----
+Thanks for contributing!
+Please keep in mind that lesson maintainers are volunteers and it may be some time before they can respond to your contribution. Although not all contributions can be incorporated into the lesson materials, we appreciate your time and effort to improve the curriculum.
```
Comment (bouweandela) [inline .github/PULL_REQUEST_TEMPLATE.md:4]: I would rephrase this a bit, e.g. propose that people submit an issue first to discuss their idea before implementing it. That way you avoid disappointment and are sure that all contributions can actually be incorporated.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 10 above)
Comment (SarahAlidoost) [reply .github/PULL_REQUEST_TEMPLATE.md]: @bouweandela My apologies. I corrected this item. The codacy checks contain a markdown linter.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 11 above)
Comment (SarahAlidoost) [reply .github/PULL_REQUEST_TEMPLATE.md]: @bouweandela thanks. I added a section "before you start" in the checklist.
